### ISSUE TITLE
JavaScript resources

### ISSUE THREAD

**Issue by katjad** [CodeHubOrg/discussions#1]
**Basics**
Codecademy JavaScript track https://www.codecademy.com/tracks/javascript
Udacity - JavaScript for Beginners https://www.udacity.com/course/javascript-basics--ud804

**General**
https://github.com/getify/You-Dont-Know-JS A book series on JavaScript. @YDKJS on twitter.

https://github.com/ericelliott/essential-javascript-links Pretty comprehensive list of resources for learning JS

"JavaScript Path" in form of books to work through. Nr 1 is Eloquent JavaScript! 
[https://github.com/javascript-society/javascript-path](https://github.com/javascript-society/javascript-path)

Bitovi Frontend Training - From May 2015 have been doing live sessions each Wednesday 6pm each week a specific topic; videos available here https://plus.google.com/+Bitovi/posts, listed by topic here http://blog.bitovi.com/training-series-element-styles-and-positioning/

**Object-Oriented JavaScript** 
Udacity course https://www.udacity.com/course/object-oriented-javascript--ud015

**Functional programming**
This is a very good video series about functional programming by Spotify developer Mattias Petter Johansson (@mpjme) Thanks for the tip, Sam!
https://www.youtube.com/watch?v=BMUiFMZr7vk&list=PL0zVEGEvSaeEd9hlmCXrk5yUyqUag-n84

Series of Videos "What makes JS weird" Concepts like scope, closures etc. quite well explained
Generally **LearnCode.academy** youtube channel ([https://www.youtube.com/user/learncodeacademy][1]) very good for JS and other web dev
https://www.youtube.com/watch?v=JEq7Ehw-qk8&index=20

**JS (and not only) reference**
Reference for everything you need for web development, including libraries and frameworks. Works offline, too
http://devdocs.io

**Test your JS**
Test your JavaScript knowledge, by making tests pass (by Rebecca Murphy) https://github.com/rmurphey/js-assessment

  [1]: https://www.youtube.com/user/learncodeacademy

**Comment by mattpill** [comment]
Hi guys,

I discussed during the javascript 101 meetup about what I've been using to teach myself coding and what I've found most useful. The best I've found so far is **treehouse**

This is not a free option (it costs $25 per month) but I've found this to be the most straight forward a structured way to learn.

You can sign up to a free 7 day trial to see if its for you (just remember to cancel it before then if its not for you, otherwise it'll go on to charge you for the month. I've left a link below which I believe gives you 50% off your first month (and 50% off my next month too I think/hope!):

http://referrals.trhou.se/mattpill

Anyway - it would be good to see how others find using it. Personally its the best thing thats worked for me anyway so hopefully it would be the same for you!

Cheers.

**Comment by wingedeel** [comment]
Hi Matt

Have tried Treehouse for the past week and have found it useful so have signed up! Thanks for the tip. I used your referral link so I hope that's worked (though I'm not entirely sure it has).

Cheers
Anne

**Comment by mattpill** [comment]
Hi Anne,

Glad you liked it too! It's definitely the best tool I've used so far to help boost my learning. We can chat about what you've worked through during the next meetup if you like.

Thanks,
Matt

**Comment by wingedeel** [comment]
Yes, would be really useful to chat about we're working on.  That JS guy is
annoying but he does tell you the important stuff it seems.

**Comment by mattpill** [comment]
Yes totally agree with the JS guy! They all seem to go at a good pace though, both for newcomers and those just wanting to go over some aspects. I'll see you at the next meetup!
